nexus/auth/src/authz/api_resources.rs

```diff
@@ -103,27 +101,27 @@ pub trait ApiResourceWithRoles: ApiResource {
 pub trait ApiResourceWithRolesType: ApiResourceWithRoles {
     type AllowedRoles: serde::Serialize
         + serde::de::DeserializeOwned
-        + db::model::DatabaseString
+        + nexus_db_model::DatabaseString
         + Clone;
 }
 
-impl<T: ApiResource + oso::PolarClass + Clone> AuthorizedResource for T {
-    fn load_roles<'a, 'b, 'c, 'd, 'e, 'f>(
+impl<T> AuthorizedResource for T
+where
+    T: ApiResource + oso::PolarClass + Clone,
+{
+    fn load_roles<'a, 'b, 'c, 'd, 'e>(
         &'a self,
         opctx: &'b OpContext,
-        datastore: &'c DataStore,
```

Comment:
You'll see this a lot in this PR: Places where we pass both `opctx` and `datastore` actually don't *need* to pass both, since `opctx` itself contains a reference to the `datastore`.

By refactoring this logic, I cut down on the number of arguments, here and elsewhere in this PR.

Since we're abstracting away access to the datastore as a `dyn Storage` trait now, this also will be easier to refactor in the future.

Reply:
Oh, lovely --- I hadn't realized `OpContext` also had a `DataStore` in it, that would have made some of my refactor-things-into-free-functions change a bit nicer. Whoops!